src/main/java/org/jenkinsci/plugins/vSphereCloudSlave.java

```diff
@@ -365,6 +367,14 @@ public void preLaunch(Computer c, TaskListener taskListener) throws IOException,
         }
     }
 
+    @Extension
+    public static class DescriptorVisibilityFilterImpl extends DescriptorVisibilityFilter {
+        @Override
+        public boolean filter(@CheckForNull Object context, @NonNull Descriptor descriptor) {
+            return !(descriptor instanceof vSphereCloudProvisionedSlave.DescriptorImpl);
```

Comment:
copy pasta 😆

Reply:
```suggestion
            return !(descriptor instanceof DescriptorImpl);
```

Reply:
I'd go with the former.  Where we've got
```
class C {
  class Inner { ... }
```
then code in class C that references class C.Inner should do so by saying `Inner` not `C.Inner`.

...but I'm still not entirely convinced that, even expressed like that, the code is doing what's needed, and the fact that this slipped through suggests that whatever testing you did, didn't pick this up.

Reply:
The only way to test it is through UI and I didn't test it since I just copy pasted ;) from one location which was the wrong thing to do

Reply:
It's been battled tested for the past year 😆

Reply:
Actually if you notice, this is the correct one implemented in `vSphereCloudProvisionedSlave` but since I copy pasted it from `vSphereCloudProvisionedSlave` to `vSphereCloudSlave`. Intellij decided to use `vSphereCloudProvisionedSlave.DescriptorImpl`

```
    @Extension
    public static class DescriptorVisibilityFilterImpl extends DescriptorVisibilityFilter {
        @Override
        public boolean filter(@CheckForNull Object context, @NonNull Descriptor descriptor) {
            return !(descriptor instanceof DescriptorImpl);
        }
    }
```

Reply:
FYI Eclipse does that sort of thing too - decides that it knows better than to paste exactly the text you copied and instead tried to edit the code so it does exactly the same things as what you copied.  99% of the time, this isn't what I want...

...and, yes, to test it requires using the GUI.  Most of the `Descriptor` code is purely there to support the GUI.
I know that your primary use-case is JCasC, but this plugin can't abandon the 99.9% of other users who use the Jenkins configuration GUI to configure Jenkins, so that means that we _must_ test this before it gets merged.

Reply:
Fixed